src/choose_block_size.jl

```diff
@@ -7,3 +7,21 @@ function choose_block_size(C, A, B, ::Nothing)
 end
 
 choose_block_size(C, A, B, block_size::Integer) = block_size
+
+function choose_parameter(C, A, B, ::Nothing)
+    m = Int128(size(A, 1))
+    n = Int128(size(B, 1))
+    k = Int128(size(A, 2))
+    oversubscription_ratio = 8  # a magic constant that Works For Me
```

Comment:
This may be platform dependent, e.g. a function of single threaded matmul size. E.g., 2-fma AVX512 may want a larger threshold.
I'm imagining that it'd lead to a lot of oversubscription on my 7980/10980XEs, for example:
```julia
julia> 20^3 <= cld(100^3, 18*8) # Threads.nthreads() == VectorizationBase.num_cores()
false

julia> 16^3 <= cld(100^3, 36*8) # Threads.nthreads() == Sys.CPU_THREADS
false
```

Reply:
Yeah, it'd be nice to find a way to set it with an informed guess, using some parameters from VectorizationBase. I decided to pass `singlethread_size` around via parameter (rather than baking in its ratio as a `const`) so that it's easier to experiment.